public/pages/Destinations/containers/CreateDestination/ManageEmailGroups/ManageEmailGroups.js

```diff
@@ -0,0 +1,261 @@
+ *   permissions and limitations under the License.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import { Formik, FieldArray } from 'formik';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiSpacer,
+} from '@elastic/eui';
+import PropTypes from 'prop-types';
+
+import AddEmailGroupButton from '../../../components/createDestinations/AddEmailGroupButton/AddEmailGroupButton';
+import EmailGroup from '../../../components/createDestinations/Email/EmailGroup';
+import EmailGroupEmptyPrompt from '../../../components/createDestinations/EmailGroupEmptyPrompt';
+import { emailGroupToFormik } from './utils/helpers';
+import getEmailGroups from '../EmailRecipients/utils/helpers';
+import { STATE } from '../../../components/createDestinations/Email/utils/constants';
+
+const createEmailGroupContext = emailGroups => ({
+  ctx: {
+    emailGroups,
+  },
+});
+
+const getInitialValues = emailGroups =>
+  _.isEmpty(emailGroups)
+    ? { emailGroups: [] }
+    : { emailGroups: emailGroups.map(emailGroup => emailGroupToFormik(emailGroup)) };
+
+const getEmailOptions = emailGroups => {
+  if (_.isEmpty(emailGroups)) return [];
+
+  // Return a unique list of all emails across all email groups
+  const nestedEmails = emailGroups.map(emailGroup => emailGroup.emails);
+  const emails = [].concat(...nestedEmails);
+  // Don't wrap the email in a label if it was a custom option since it already is
+  return [...new Set(emails)].map(email => (_.isString(email) ? { label: email } : email));
+};
+
+export default class ManageEmailGroups extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      emailGroupsToDelete: [],
+      loadingEmailGroups: true,
+    };
+
+    this.getEmailGroups = getEmailGroups.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadInitialValues();
+  }
+
+  // Reload initial values when modal is no longer visible so changes
+  // are reflected the next time it is opened
+  componentDidUpdate(prevProps) {
+    if (prevProps.isVisible && !this.props.isVisible) {
+      this.loadInitialValues(this.props.emailGroups);
+    }
+  }
+
+  loadInitialValues = async () => {
+    this.setState({ loadingEmailGroups: true });
+
+    const emailGroups = await this.getEmailGroups();
+    const initialValues = getInitialValues(emailGroups);
+
+    this.setState({
+      initialValues,
+      emailGroupsToDelete: [],
+      loadingEmailGroups: false,
+    });
+  };
+
+  createEmailGroup = async emailGroup => {
+    const { httpClient } = this.props;
+    const body = {
+      name: emailGroup.name,
+      emails: emailGroup.emails.map(email => ({ email: email.label })),
+    };
+    try {
+      await httpClient.post(`../api/alerting/email_groups`, body);
+    } catch (err) {
+      console.error('Unable to create email group', err);
```

Comment:
What is UX behavior here? No feedback happens on UI? Same question for other ones.

Reply:
Added toast messages on success and failures for both modals.